test/OpenTelemetry.Instrumentation.EntityFrameworkCore.Tests/EntityFrameworkDiagnosticListenerTests.cs

```diff
@@ -70,6 +70,42 @@ public void EntityFrameworkContextEventsInstrumentedTest()
         VerifyActivityData(activity);
     }
 
+    [Fact]
+    public void EntityFrameworkContextWithAlternativeDisplayName()
+    {
+        var activityProcessor = new Mock<BaseProcessor<Activity>>();
+        var expectedCommand = "SELECT \"i\".\"Id\", \"i\".\"Name\"" +
+                              "\nFROM \"Item\" AS \"i\"" +
+                              "\nORDER BY \"i\".\"Name\" main";
+        using var shutdownSignal = Sdk.CreateTracerProviderBuilder()
+            .AddProcessor(activityProcessor.Object)
+            .AddEntityFrameworkCoreInstrumentation(options =>
+            {
+                options.EnrichWithIDbCommand = (activity1, command) =>
+                {
+                    var stateDisplayName = $"{command.CommandText} main";
+                    activity1.DisplayName = stateDisplayName;
+                    activity1.SetTag("db.name", stateDisplayName);
+                };
+            }).Build();
+
+        using (var context = new ItemsContext(this.contextOptions))
+        {
+            var items = context.Set<Item>().OrderBy(e => e.Name).ToList();
+
+            Assert.Equal(3, items.Count);
+            Assert.Equal("ItemOne", items[0].Name);
+            Assert.Equal("ItemThree", items[1].Name);
+            Assert.Equal("ItemTwo", items[2].Name);
+        }
+
+        Assert.Equal(3, activityProcessor.Invocations.Count);
+
+        var activity = (Activity)activityProcessor.Invocations[1].Arguments[0];
+
+        VerifyActivityData(activity, altDisplayName: $"{expectedCommand.Replace("\r\n", "\n")}");
```

Comment:
There's still a test failure here

>   Failed OpenTelemetry.Instrumentation.EntityFrameworkCore.Tests.EntityFrameworkDiagnosticListenerTests.EntityFrameworkContextWithAlternativeDisplayName [76 ms]
  Error Message:
   Assert.Equal() Failure
                                 ↓ (pos 27)
Expected: ···"i"."Id", "i"."Name"\nFROM "Item" AS "i"\nORDER BY "i"."Name" m···
Actual:   ···"i"."Id", "i"."Name"\r\nFROM "Item" AS "i"\r\nORDER BY "i"."Name"···
                                 ↑ (pos 27)

Is using the command text in the alternate name relevant to your test? Could you just override the name to something simple like `othername`?

Reply:
Updated now, main reason I wanted to use the dbcommand itself was to fully verify access to the command. I've simplified it now to use the command type.